constantine/ethereum_verkle_ipa.nim

```diff
@@ -137,10 +157,19 @@ template check(Section: untyped, evalExpr: CttCodecEccStatus): untyped {.dirty.}
 # ------------------------------------------------------------------------------------
 
 type
-  EthVerkleIpaProofBytes* = array[544, byte]
-  EthVerkleIpaMultiProofBytes* = array[576, byte]
-  EthVerkleIpaProof* = IpaProof[8, EC_TwEdw[Fp[Banderwagon]], Fr[Banderwagon]]
-  EthVerkleIpaMultiProof* = IpaMultiProof[8, EC_TwEdw[Fp[Banderwagon]], Fr[Banderwagon]]
+  EthVerkleIpaProofBytes* {.exportc: prefix_ffi & "proof_bytes".}= array[544, byte]
+  EthVerkleIpaMultiProofBytes* {.exportc: prefix_ffi & "multi_proof_bytes".}= array[576, byte]
+  EthVerkleIpaProof* {.exportc: prefix_ffi & "proof".} = object
+    aff: IpaProof[8, EC_TwEdw_Aff[Fp[Banderwagon]], Fr[Banderwagon]]
+    prj: IpaProof[8, EC_TwEdw_Prj[Fp[Banderwagon]], Fr[Banderwagon]]
+  EthVerkleIpaMultiProof* {.exportc: prefix_ffi & "multi_proof".} = object
+    aff: IpaMultiProof[8, EC_TwEdw_Aff[Fp[Banderwagon]], Fr[Banderwagon]]
+    prj: IpaMultiProof[8, EC_TwEdw_Prj[Fp[Banderwagon]], Fr[Banderwagon]]
+  EthVerkleIpaPolyEvalCrs* {.exportc: prefix_ffi & "poly_eval".} = object
+    aff: PolynomialEval[EthVerkleDomain, EC_TwEdw_Aff[Fp[Banderwagon]]]
+    prj: PolynomialEval[EthVerkleDomain, EC_TwEdw_Prj[Fp[Banderwagon]]]
+  EthVerkleIpaPolyEvalPoly* {.exportc: prefix_ffi & "poly_eval_poly".} = object
+    raw: PolynomialEval[256, Fr[Banderwagon]]
```

Comment:
The file should be reverted and not modified.

This approach doubles object size while we only need aff or prj coordinates. The serialized code should just use `EthVerkleIpaProofBytes` and unsure yet on the C API of the internal proof type, maybe will hide it behind an opaque pointer.